src/main/java/service/FileWriterService.java

```diff
@@ -0,0 +1,7 @@
+package service;
+
+import java.util.Map;
+
+public interface FileWriterService {
+    void writeToFile(Map<String, Integer> map);

```

Comment:
no business logic allowed in IO related files. you can use report string as parameter as suggested or list of strings
```suggestion
    void writeToFile(String content);
```